Compiling with external hwloc results in many warnings

After #5395 and #5466, if you compile with an external hwloc, there are many warnings:

```
/home/jsquyres/bogus/include/hwloc/autogen/config.h:214:1: warning: "HWLOC_SYM_TRANSFORM" redefined
In file included from error.c:26:
../../opal/include/opal_config.h:1394:1: warning: this is the location of the previous definition
In file included from /home/jsquyres/bogus/include/hwloc.h:56,
                 from ../../opal/mca/hwloc/external/external.h:24,
                 from ../../opal/mca/hwloc/hwloc-internal.h:131,
                 from ../../opal/util/proc.h:22,
                 from error.c:36:
/home/jsquyres/bogus/include/hwloc/autogen/config.h:217:1: warning: "HWLOC_SYM_PREFIX" redefined
In file included from error.c:26:
../../opal/include/opal_config.h:1388:1: warning: this is the location of the previous definition
In file included from /home/jsquyres/bogus/include/hwloc.h:56,
                 from ../../opal/mca/hwloc/external/external.h:24,
                 from ../../opal/mca/hwloc/hwloc-internal.h:131,
                 from ../../opal/util/proc.h:22,
                 from error.c:36:
/home/jsquyres/bogus/include/hwloc/autogen/config.h:220:1: warning: "HWLOC_SYM_PREFIX_CAPS" redefined
In file included from error.c:26:
../../opal/include/opal_config.h:1391:1: warning: this is the location of the previous definition
  CC       info.lo
In file included from /home/jsquyres/bogus/include/hwloc.h:56,
                 from ../../opal/mca/hwloc/external/external.h:24,
                 from ../../opal/mca/hwloc/hwloc-internal.h:131,
                 from proc.h:22,
                 from proc.c:19:
```

I just build from the hwloc master head:

```
$ cd path/to/hwloc/repo
$ git show | head -n 5
commit 34f67eac480d9ae13a98b45da6a116f17f7c59ce
Author: Brice Goglin <Brice.Goglin@inria.fr>
Date:   Mon Aug 6 21:43:16 2018 +0200

    doxy: cleanup/fix the object that may contain some hwloc-provided info attributes
$ ./autogen.sh |& tee auto.out && ./configure --prefix=$bogus|& tee config.out && make clean > /dev/null && rm -rf $bogus && j32 |& tee make.out && make install > /dev/null
...

$ cd path/to/ompi/repo
$ git show | head -n 6
commit e9f378e8518fa7a0a17586f93ae616b50f2d3713
Merge: c294bbc352 f3f2a826b4
Author: Todd Kordenbrock <tkordenbrock@users.noreply.github.com>
Date:   Tue Aug 7 11:21:00 2018 -0500

    Merge pull request #5500 from tkordenbrock/topic/master/fix.PtlMEUnlink.in.use
$ ./autogen.pl|& tee auto.out && ./configure --prefix=$bogus CPPFLAGS=-I$bogus/include LDFLAGS=-L$bogus/lib |& tee config.out && make clean > /dev/null && j32 |& tee make.out
...
/home/jsquyres/bogus/include/hwloc/autogen/config.h:214:1: warning: "HWLOC_SYM_TRANSFORM" redefined
In file included from error.c:26:
../../opal/include/opal_config.h:1394:1: warning: this is the location of the previous definition
In file included from /home/jsquyres/bogus/include/hwloc.h:56,
                 from ../../opal/mca/hwloc/external/external.h:24,
                 from ../../opal/mca/hwloc/hwloc-internal.h:131,
                 from ../../opal/util/proc.h:22,
                 from error.c:36:
...etc.
```

Perhaps it's because I'm using an old compiler -- gcc 4.4.7...?  

...time passes...

Nope -- same thing happens if I test with gcc 7.3.0, too.

--- a/opal/mca/event/libevent2022/configure.m4
+++ b/opal/mca/event/libevent2022/configure.m4
@@ -1,6 +1,6 @@
 # -*- shell-script -*-
 #
-# Copyright (c) 2009-2015 Cisco Systems, Inc.  All rights reserved.
+# Copyright (c) 2009-2018 Cisco Systems, Inc.  All rights reserved.
 # Copyright (c) 2012-2013 Los Alamos National Security, LLC.  All rights reserved.
 # Copyright (c) 2015      Intel, Inc. All rights reserved.
 # Copyright (c) 2015-2016 Research Organization for Information Science
@@ -86,6 +86,21 @@ EOF
 #                              [action-if-cant-compile])
 # ------------------------------------------------
 AC_DEFUN([MCA_opal_event_libevent2022_CONFIG],[
+    # We know that the external event component will be configured
+    # before this one because of its priority.  This component is only
+    # needed if the external component was not successful in selecting
+    # itself.
+    AC_MSG_CHECKING([if event external component succeeded])
+    AS_IF([test "$opal_event_external_support" = "yes"],
+          [AC_MSG_RESULT([yes])
+           AC_MSG_NOTICE([event:external succeeded, so this component will be skipped])
+           $2],
+          [AC_MSG_RESULT([no])
+           AC_MSG_NOTICE([event:external failed, so this component will be used])
+           MCA_opal_event_libevent2022_BACKEND_CONFIG($1, $2)])
+])
+
+AC_DEFUN([MCA_opal_event_libevent2022_BACKEND_CONFIG],[
     OPAL_VAR_SCOPE_PUSH([CFLAGS_save CPPFLAGS_save libevent_file event_args libevent_happy])
 
     AC_CONFIG_FILES([opal/mca/event/libevent2022/Makefile])

--- a/opal/mca/hwloc/hwloc201/configure.m4
+++ b/opal/mca/hwloc/hwloc201/configure.m4
@@ -1,6 +1,6 @@
 # -*- shell-script -*-
 #
-# Copyright (c) 2009-2017 Cisco Systems, Inc.  All rights reserved
+# Copyright (c) 2009-2018 Cisco Systems, Inc.  All rights reserved
 # Copyright (c) 2014-2017 Intel, Inc. All rights reserved.
 # Copyright (c) 2015-2018 Research Organization for Information Science
 #                         and Technology (RIST). All rights reserved.
@@ -69,6 +69,21 @@ AC_DEFUN([MCA_opal_hwloc_hwloc201_POST_CONFIG],[
 # MCA_hwloc_hwloc201_CONFIG([action-if-found], [action-if-not-found])
 # --------------------------------------------------------------------
 AC_DEFUN([MCA_opal_hwloc_hwloc201_CONFIG],[
+    # We know that the external hwloc component will be configured
+    # before this one because of its priority.  This component is only
+    # needed if the external component was not successful in selecting
+    # itself.
+    AC_MSG_CHECKING([if hwloc external component succeeded])
+    AS_IF([test "$opal_hwloc_external_support" = "yes"],
+          [AC_MSG_RESULT([yes])
+           AC_MSG_NOTICE([hwloc:external succeeded, so this component will be skipped])
+           $2],
+          [AC_MSG_RESULT([no])
+           AC_MSG_NOTICE([hwloc:external failed, so this component will be used])
+           MCA_opal_hwloc_hwloc201_BACKEND_CONFIG($1, $2)])
+])
+
+AC_DEFUN([MCA_opal_hwloc_hwloc201_BACKEND_CONFIG],[
     # Hwloc needs to know if we have Verbs support
     AC_REQUIRE([OPAL_CHECK_VERBS_DIR])
 
@@ -80,19 +95,11 @@ AC_DEFUN([MCA_opal_hwloc_hwloc201_CONFIG],[
     opal_hwloc_hwloc201_basedir=opal/mca/hwloc/hwloc201
     opal_hwloc_hwloc201_support=no
 
-    AS_IF([test "$with_hwloc" = "internal" || test -z "$with_hwloc" || test "$with_hwloc" = "yes"],
-          [opal_hwloc_external="no"],
-          [opal_hwloc_external="yes"])
-
     opal_hwloc_hwloc201_save_CPPFLAGS=$CPPFLAGS
     opal_hwloc_hwloc201_save_LDFLAGS=$LDFLAGS
     opal_hwloc_hwloc201_save_LIBS=$LIBS
 
-    # Run the hwloc configuration - if no external hwloc, then set the prefixi
-    # to minimize the chance that someone will use the internal symbols
-    AS_IF([test "$opal_hwloc_external" = "no" &&
-           test "$with_hwloc" != "future"],
-          [HWLOC_SET_SYMBOL_PREFIX([opal_hwloc201_])])
+    HWLOC_SET_SYMBOL_PREFIX([opal_hwloc201_])
 
     # save XML or graphical options
     opal_hwloc_hwloc201_save_cairo=$enable_cairo

